caffe2-haskell/cpu/Foreign/Caffe2.hs

```diff
@@ -99,17 +101,18 @@ readCPUBlob = readBlob
 #define WRITE_TENSOR(BLOB,CTYPE,VEC,SIZE,DIMS)                                           \
  Vector.unsafeWith (VEC) $ \ptr -> [C.block| void {                                      \
     Blob *blob = static_cast<Blob *>($(void *BLOB));                                     \
-    int64_t *arrInputDims = $vec-ptr:(int64_t *DIMS);                                    \
-    std::vector<int64_t> shape =                                                         \
-      std::vector<int64_t>(arrInputDims, arrInputDims + $vec-len:DIMS);                  \
-    auto *tensor = BlobGetMutableTensor(blob, CPU);                                      \
-    tensor->Resize(shape);                                                               \
+    int *arrInputDims = $vec-ptr:(int *DIMS);                                    \
+    std::vector<int> shape =                                                         \
+      std::vector<int>(arrInputDims, arrInputDims + $vec-len:DIMS);                  \
+    auto *tensor = blob->GetMutable<ideep::tensor>();                                    \
+    tensor->resize(shape, ideep::tensor::data_type::f32);                                                               \
```

Comment:
This shouldn't be hard-coded for f32, but instead take the data type here as an argument to WRITE_TENSOR.